### DISCUSSION TITLE
URL to include Directus module in iframe

### DISCUSSION THREAD

**Discussion by mattbab** [directus/directus#12939] [Feature Requests]
I'd like to be able to expose a Directus module or collection as part of a larger system. I have another Vue application in which I'd like to render the Directus module or collection within an iframe. This works ok after overcoming the baked in CSP restrictions, but is suboptimal when rendered next to a "parent-app" navigation since Directus provides its own navigation.

The ideal implementation would allow me to provide an optional query string to hide the module navigation. It would also be acceptable to be able to switch the side the navigation appears on if that's simple. I may have missed it while reviewing the code, but neither option appears to be possible in the current implementation.

While reviewing the extensions APIs it seems this may be possible by implementing a custom layout since it is provided an instance of the module bar. Before diving into writing such an extension, I wanted to, first rule out that it's not otherwise possible, second, if the community can provide any explanation why this wouldn't work, and finally, if my understanding of the layout extension documentation page is correct.

Thanks!

**Comment by rijkvanzanten** [comment]
Directus isn't currently designed/built to work in an `iframe` context, so your mileage may vary.. When it comes to custom extensions, a custom Module allows you to render whatever you want (you don't even have to use the page chrome if you don't want), so that could be a starting point. That being said, seeing that nothing else was designed with this use case in mind, you might run into some other similar UX problems 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
